app/namespaces/test_adjustment/test_adjustment_service.py

```diff
@@ -0,0 +1,262 @@
+            return None, None, None
+
+        satisfactory_model_set_found = False
+        n_modelsets = 0
+
+        while not satisfactory_model_set_found:
+            satisfactory_model_set = []
+            for _ in range(self.n_replicates):
+                n_trials = 0
+                satisfactory_model_found = False
+
+                while not satisfactory_model_found:
+                    # Attempt to fit a model
+                    summary_df_parsed, satisfactory_model_found = self.fit_one_pystan_model(model_params)
+                    # If number of attempts exceeded, return Nones
+                    n_trials += 1
+                    if n_trials >= self.trials_lim:
+                        print('no models met the HMC diagnostics in {trials_lim} trials')
+                        return None, None, None
+
+                satisfactory_model_set.append(summary_df_parsed)
+
+            satisfactory_models_df = pd.DataFrame(satisfactory_model_set)
+
+            # see whether all results are within 10% of median; if not, rerun
+            model_medians = satisfactory_models_df['50%']
+            median = model_medians.median()
+            consistent_results_found = model_medians.between(0.9 * median, 1.1 * median).all()
+            median_run = satisfactory_models_df[satisfactory_models_df['50%'] == median].iloc[0]
+
+            lower, upper = arviz.hdi(median_run['samples'], credible_interval_size)
+            best_fit = median_run['fit']
+
+            try:
+                raw_prev = model_params['y_prev_obs'] / model_params['n_prev_obs']
+                bounded = result_is_bounded(median, raw_prev)
+            except (ZeroDivisionError, ValueError):
+                return None, None, None
+
+            satisfactory_model_set_found = consistent_results_found and bounded
+
+            n_modelsets += 1
+            if n_modelsets > self.modelsets_lim:
+                return None, None, None
+
+        if self.return_fit:
+            return best_fit
+        else:
+            return lower, median, upper
+
+    def get_adjusted_estimate(self, test_adj, ind_se, ind_sp, ind_se_n, ind_sp_n,
+                              se_n, sp_n, sensitivity, specificity, test_validation,
+                              test_type, denominator_value, serum_pos_prevalence) -> Tuple:
+        # initialize adj_type
+        adj_type = None
+
+        # unadjusted estimate available and thus prioritized in estimate selection code
+        if pd.isna(test_adj):
+
+            # Independent evaluation is available
+            if pd.notna(ind_se) and pd.notna(ind_sp):
+                adj_type = 'FINDDx / MUHC independent evaluation'
+                # Also note these must be divided by 100
+                se = ind_se
+                sp = ind_sp
+                output_se_n = float(ind_se_n) * 100 if ind_se_n is not None else 30
+                output_sp_n = float(ind_sp_n) * 100 if ind_sp_n is not None else 80
+
```

Comment:
Note that the research team is working on adding other independent evaluation sources to Airtable and updating how we want to prioritize independent, author, and manufacturer evaluation. Will let you know when this is in place